### PR TITLE
Add Ctrl(+Shift)+Tab as alternative to switch between servers

### LINKED ISSUE #512: Ctrl+Tab should switch between servers when logged into multiple servers
I confirm (by marking "x" in the [ ] below): 

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**
Pressing Ctrl+Tab should switch tabs when logged into multiple teams.

**Steps to reproduce**
1. Open Mattermost Desktop
2. Sign in to at least two servers, creating multiple tabs across the top of the app
3. Press Ctrl+Tab
4. Observe that nothing happens.

- Operating System: macOS 10.12.4
- Mattermost Desktop App version: 3.6.0
- Mattermost Server version: 3.7.3

**Expected behavior**
I should switch between tabs, similar to how other common tabbed interfaces work (Chrome, FF, Visual Studio, etc.)

**Observed behavior**
Nothing happens when you press Ctrl+Tab

### PR DESCRIPTION
**Pull request by wvds**
Closes #512

Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
[Mac] Adds `Ctrl+Tab` and `Ctrl+Shift+Tab` keyboard shortcuts as an alternative to `Alt+Cmd+Right` and `Alt+Cmd+Left` to switch between servers.

**Issue link**
#512

### CODE REVIEW

File: src/main/menus/app.js [hunk 1]
```diff
@@ -186,14 +186,14 @@ function createTemplate(mainWindow, config, isDev) {
       };
     }), separatorItem, {
       label: 'Select Next Server',
-      accelerator: (process.platform === 'darwin') ? 'Alt+Cmd+Right' : 'CmdOrCtrl+Tab',
+      accelerator: 'Ctrl+Tab',
```
Comment (yuya-oc) [inline src/main/menus/app.js]: I had thought `Alt+Cmd+Right` is appropriate here (See Chrome's menu), but Safari uses `Ctrl+Tab`. I think it's more consistent on macOS 👍

File: src/main.js [hunk 2]
```diff
@@ -342,6 +343,17 @@ app.on('ready', () => {
     console.log('The application has crashed.');
   });
 
+  // Add Alt+Cmd+(Right|Left) as alternative to switch between servers
+  if (process.platform === 'darwin') {
+    globalShortcut.register('Alt+Cmd+Right', () => {
+      mainWindow.webContents.send('select-next-tab');
+    });
+
+    globalShortcut.register('Alt+Cmd+Left', () => {
```
Comment (yuya-oc) [inline src/main.js]: Do not use globalShortcut. You can use a  MenuItem with `visible: false` parameter.

File: src/main.js [hunk 3]
```diff
@@ -342,6 +343,17 @@ app.on('ready', () => {
     console.log('The application has crashed.');
   });
 
+  // Add Alt+Cmd+(Right|Left) as alternative to switch between servers
+  if (process.platform === 'darwin') {
+    globalShortcut.register('Alt+Cmd+Right', () => {
```
Comment (yuya-oc) [inline src/main.js]: Do not use globalShortcut. You can use a  MenuItem with `visible: false` parameter.

Comment (wvds) [reply src/main.js]: Unfortunately this solution isn't working on Mac, and if it's still working for Windows it's a bug, see: https://github.com/electron/electron/issues/7737.

I've updated the PR with the `before-input-event` implementation which is mentioned in that thread.

File: src/main.js (hunk 2 above)
Comment (wvds) [reply src/main.js]: Unfortunately this solution isn't working on Mac, and if it's still working for Windows it's a bug, see: https://github.com/electron/electron/issues/7737.

I've updated the PR with the `before-input-event` implementation which is mentioned in that thread.

**Comment by yuya-oc** [comment]
Nice feedback, @wvds! I will take that way. And we should consider moving hidden accelerators to `before-input-event` in future.

https://circleci.com/gh/mattermost/desktop/1058#artifacts

**Comment by jasonblais** [comment]
Have asked for a team member to help test this out on Mac

**Comment by lindalumitchell** [comment]
@jasonblais @wvds I've tested this on MacOS, and looks great! No issues found.

**Comment by jasonblais** [comment]
Thanks @lindalumitchell! This should be good to merge @yuya-oc 

Big thanks @wvds :)

File: src/main.js [hunk 4]
```diff
@@ -342,6 +342,20 @@ app.on('ready', () => {
     console.log('The application has crashed.');
   });
 
+  // Add Alt+Cmd+(Right|Left) as alternative to switch between servers
```
Comment (yuya-oc) [inline src/main.js]: @wvds Would you move this part into `src/main/mainWindow.js`? It would make sense when adding further hidden shortcuts.

Comment (wvds) [reply src/main.js]: I've updated the pull request :)

**Comment by yuya-oc** [comment]
Thanks @wvds!

**Pull request merged**